bundles/admin/admin-layereditor/view/AdminLayerFormService.js

```diff
@@ -0,0 +1,238 @@
+                me.notify();
+            },
+            setMapLayerGroup (checked, group) {
+                const layer = { ...me.layer };
+                if (checked) {
+                    layer.maplayerGroups = [...layer.maplayerGroups, group];
+                } else {
+                    const found = layer.maplayerGroups.find(cur => cur.id === group.id);
+                    if (found) {
+                        layer.maplayerGroups = [...layer.maplayerGroups];
+                        layer.maplayerGroups.splice(layer.maplayerGroups.indexOf(found), 1);
+                    }
+                }
+                me.layer = layer;
+                me.notify();
+            },
+            setOpacity (opacity) {
+                me.layer = { ...me.layer, opacity };
+                me.notify();
+            },
+            setMinAndMaxScale (values) {
+                me.layer = {
+                    ...me.layer,
+                    maxScale: values[0],
+                    minScale: values[1]
+                };
+                me.notify();
+            },
+            setStyle (style) {
+                me.layer = { ...me.layer, style };
+                me.notify();
+            },
+            setStyleJSON (styleJSON) {
+                me.layer = { ...me.layer, styleJSON };
+                me.notify();
+            },
+            setHoverJSON (hoverJSON) {
+                me.layer = { ...me.layer, hoverJSON };
+                me.notify();
+            },
+            setMetadataIdentifier (metadataIdentifier) {
+                me.layer = { ...me.layer, metadataIdentifier };
+                me.notify();
+            },
+            setGfiContent (gfiContent) {
+                me.layer = { ...me.layer, gfiContent };
+                me.notify();
+            },
+            setAttributes (attributes) {
+                me.layer = { ...me.layer, attributes };
+                me.notify();
+            },
+            setMessage (key, type) {
+                me.message = {
+                    key: key,
+                    type: type
+                };
+                me.notify();
+            }
+        };
+    }
+
+    initLayerState (layer) {
+        var me = this;
+
+        const styles = layer ? layer.getStyles() : [];
+        const availableStyles = [];
+        for (let i = 0; i < styles.length; i++) {
+            availableStyles.push({
+                name: styles[i].getName(),
+                title: styles[i].getTitle()
+            });
+        }
+
+        me.layer = {
+            version: layer ? layer.getVersion() : '',
+            layer_id: layer ? layer.getId() : null,
+            layerUrl: layer ? layer.getAdmin().url : '',
+            username: layer ? layer.getAdmin().username : '',
+            password: layer ? layer.getAdmin().password : '',
+            layerName: layer ? layer.getLayerName() : '',
+            name_fi: layer ? layer.getName('fi') : '',
```

Comment:
This won't work as we have other locales as well but let's worry about it later.

Reply:
Added TODO comments so we won't forget this issue.